src/main/java/net/earthcomputer/clientcommands/command/ListenCommand.java

```diff
@@ -0,0 +1,269 @@
+package net.earthcomputer.clientcommands.command;
+
+import com.google.gson.stream.JsonWriter;
+import com.mojang.brigadier.Command;
+import com.mojang.brigadier.CommandDispatcher;
+import com.mojang.brigadier.Message;
+import com.mojang.brigadier.exceptions.CommandSyntaxException;
+import com.mojang.brigadier.exceptions.SimpleCommandExceptionType;
+import it.unimi.dsi.fastutil.objects.ReferenceOpenHashSet;
+import it.unimi.dsi.fastutil.objects.ReferenceSet;
+import net.earthcomputer.clientcommands.*;
+import net.earthcomputer.clientcommands.features.MappingsHelper;
+import net.earthcomputer.clientcommands.features.PacketDumper;
+import net.fabricmc.fabric.api.client.command.v2.FabricClientCommandSource;
+import net.minecraft.ChatFormatting;
+import net.minecraft.core.Holder;
+import net.minecraft.core.Registry;
+import net.minecraft.network.chat.ClickEvent;
+import net.minecraft.network.chat.Component;
+import net.minecraft.network.chat.HoverEvent;
+import net.minecraft.network.chat.MutableComponent;
+import net.minecraft.network.protocol.Packet;
+import net.minecraft.network.protocol.PacketFlow;
+import net.minecraft.resources.ResourceKey;
+import net.minecraft.resources.ResourceLocation;
+import net.minecraft.world.level.ChunkPos;
+
+import java.io.IOException;
+import java.io.StringWriter;
+import java.lang.invoke.VarHandle;
+import java.lang.reflect.Array;
+import java.lang.reflect.InaccessibleObjectException;
+import java.lang.reflect.Modifier;
+import java.time.Instant;
+import java.util.*;
+
+import static net.earthcomputer.clientcommands.command.arguments.MojmapPacketClassArgumentType.*;
+import static net.fabricmc.fabric.api.client.command.v2.ClientCommandManager.*;
+
+public class ListenCommand {
+
+    private static final SimpleCommandExceptionType ALREADY_LISTENING_EXCEPTION = new SimpleCommandExceptionType(Component.translatable("commands.clisten.add.failed"));
+    private static final SimpleCommandExceptionType NOT_LISTENING_EXCEPTION = new SimpleCommandExceptionType(Component.translatable("commands.clisten.remove.failed"));
+
+    private static final Set<Class<Packet<?>>> packets = new HashSet<>();
+
+    private static PacketCallback callback;
+
+    private static final ThreadLocal<ReferenceSet<Object>> SEEN = ThreadLocal.withInitial(ReferenceOpenHashSet::new);
```

Comment:
It's not necessarily bad, but why the use of fastutil here over a regular `HashSet`?

Reply:
Because that will perform the hash of the object, which could be really slow. `ReferenceSet`s use the identity hash.